src/queue.cpp

```diff
@@ -393,6 +393,22 @@ void cvk_command_kernel::update_global_push_constants(
                            VK_SHADER_STAGE_COMPUTE_BIT, pc->offset, pc->size,
                            &m_ndrange.gws);
     }
+    for (auto& md : m_kernel->get_image_channel_metadata()) {
+        auto order_offset = md.second.first;
+        auto data_type_offset = md.second.second;
+        if (order_offset != IMAGE_METADATA_EMPTY) {
+            vkCmdPushConstants(command_buffer, m_kernel->pipeline_layout(),
+                               VK_SHADER_STAGE_COMPUTE_BIT, order_offset,
+                               sizeof(uint32_t),
+                               &m_argument_values->pod_data()[order_offset]);
+        }
+        if (data_type_offset != IMAGE_METADATA_EMPTY) {
+            vkCmdPushConstants(
+                command_buffer, m_kernel->pipeline_layout(),
+                VK_SHADER_STAGE_COMPUTE_BIT, data_type_offset, sizeof(uint32_t),
+                &m_argument_values->pod_data()[data_type_offset]);
+        }
+    }
```

Comment:
Maybe this could be rewritten as a single push constant update for the entire range that contains image metadata? This would be in keeping with the pattern above, facilitating further shortening and likely faster.

Reply:
Your comment feels weird, as the code pushing pod args is not how you explain here.
I have modified the image_metadata implementation as well as the pod arg to be aggregated into one call to pushConstant.
Tell me if it was not what you were expecting.

Reply:
I'm not confident about what I have done here.
Maybe I should have at least 2 pushconstants, one for the args, one for the image metadata.
Otherwise it feels like the other information (global_size, num_workgroups) could be overridden by this last pushconstant.
What do you think?

Reply:
I have reverted this part to have no modifications regarding the args pushconstants